src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -87,14 +88,14 @@ private void EnsureWorkerIsRunning()
                     // Create pipe used to forcefully wake up curl_multi_wait calls when something important changes.
                     // This is created here rather than in Process so that the pipe is available immediately
                     // for subsequent queue calls to use.
-                    Debug.Assert(_wakeupRequestedPipeFd == 0, "Read pipe should have been cleared");
-                    Debug.Assert(_requestWakeupPipeFd == 0, "Write pipe should have been cleared");
+                    Debug.Assert(_wakeupRequestedPipeFd == null, "Read pipe should have been cleared");
+                    Debug.Assert(_requestWakeupPipeFd == null, "Write pipe should have been cleared");
                     unsafe
                     {
                         int* fds = stackalloc int[2];
                         while (Interop.CheckIo(Interop.Sys.Pipe(fds))) ;
```

Comment:
I skipped this one at first because I didn't think it would add value, and would only add complexity to the shim code.

But thinking more about it, if we are going to say all file descriptors are marshaled as intptr_t, then consistency is the value added.  I can change it.